application/models/item.go

```
@@ -1109,3 +1110,16 @@ func (i *Item) NewHistory(ctx context.Context, action string, fieldUpdate FieldU
 		NewValue:  fmt.Sprintf("%s", fieldUpdate.NewValue),
 	}
 }
+
+// FindItemsIncorrectlyRenewed locates any items that were incorrectly renewed for another year of coverage. These are
+// identified as items that are marked as paid through the year but have an earlier coverage_end_date.
+func (i *Items) FindItemsIncorrectlyRenewed(tx *pop.Connection, date time.Time) error {
+	year := date.Year()
+	firstDayOfYear := time.Date(year, 1, 1, 0, 0, 0, 0, time.UTC)
+
+	err := tx.Where("paid_through_year >= ?", year).Where("coverage_end_date < ?", firstDayOfYear).All(i)
```

Would it be worth including the maxResultSize in the query?
I guess this is how you would do it.
https://gobuffalo.io/documentation/database/querying/#join-query

If so, maybe make the page size one more than maxResultSize, in order to make it easy to test whether we should return an error.

---

Not sure if that helps much, but an interesting idea.